File: frontend/src/features/report/models/ReportResponseModel.ts
export interface ReportResponseModel {
  reportId: string;
  appointmentId: string;
  appointmentDate: string;
  appointmentStatus: string; // SCHEDULED, COMPLETED, CANCELLED
  
  // Customer info
  customerId: string;
  customerFirstName: string;
  customerLastName: string;
  customerPhone: string | null;
  
  // Technician info
  technicianId: string;
  technicianFirstName: string;
  technicianLastName: string;
  
  // Job info
  jobName: string;
  hourlyRate: number;
  
  // Work details
  hoursWorked: number;
  frais: number;
  fraisDeplacement: number;
  
  // Parts
  parts: ReportPartResponseModel[];
  
  // Calculated amounts
  laborCost: number;
  partsCost: number;
  subtotal: number;
  tpsAmount: number;
  tvqAmount: number;
  total: number;
  
  // Timestamps
  createdAt: string;
  updatedAt: string | null;
}

export interface ReportPartResponseModel {
  partId: string;
  partName: string;
  quantity: number;
  price: number;
  totalPrice: number;
  notes?: string | null;
}

File: backend/src/main/java/com/profroid/profroidapp/reportsubdomain/presentationLayer/ReportResponseModel.java
package com.profroid.profroidapp.reportsubdomain.presentationLayer;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ReportResponseModel {

    private String reportId;

    // Appointment information
    private String appointmentId;
    private String appointmentDate;
    private String appointmentStatus; // SCHEDULED, COMPLETED, CANCELLED

    // Customer information (auto-populated from appointment)
    private String customerId;
    private String customerFirstName;
    private String customerLastName;
    private String customerPhone;

    // Technician information
    private String technicianId;
    private String technicianFirstName;
    private String technicianLastName;

    // Job information
    private String jobName;
    private BigDecimal hourlyRate;

    // Work details
    private BigDecimal hoursWorked;
    private BigDecimal frais;
    private BigDecimal fraisDeplacement;

    // Parts used
    private List<ReportPartResponseModel> parts;

    // Calculated amounts
    private BigDecimal laborCost; // hoursWorked * hourlyRate
    private BigDecimal partsCost; // Sum of all parts
    private BigDecimal subtotal; // laborCost + frais + fraisDeplacement + partsCost
    private BigDecimal tpsAmount; // 5%
    private BigDecimal tvqAmount; // 9.975%
    private BigDecimal total; // subtotal + taxes

    // Timestamps
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @Data
    @Builder
    @AllArgsConstructor
    @NoArgsConstructor
    public static class ReportPartResponseModel {
        private String partId;
        private String partName;
        private Integer quantity;
        private BigDecimal price; // Manual price set by technician
        private BigDecimal totalPrice; // quantity * price
        private String notes;
    }
}
